### PR TITLE
feat: :sparkles: Improve `eds.history` component with `eds.dates`

### PR DESCRIPTION
**Pull request by Aremaki**
Improve the `eds.history` component by taking into account the date extracted from `eds.dates` component:

- It removes the medical hsitory tag for extracted entity assocaited with a recent date.
- It tags extracted entity assocaited with an old date.

Note: The threshold can be set as a parameter.

- Add the `use_dates` configuration to the `eds.history` component.
- Small fix on the ``eds.sections`` component (remove the warning when ``eds.noramalizer`` is in the pipe).
- Improve the test for eds.reason (100% coverage.).
- Improve the termynal extention for mkdocs with a pop up indication when you click on the copy icon.

## Checklist

- [x] If this PR is a bug fix, the bug is documented in the test suite.
- [x] Changes were documented in the changelog (pending section).
- [x] If necessary, changes were made to the documentation (eg new pipeline).

### CODE REVIEW

File: edsnlp/pipelines/qualifiers/history/history.py [hunk 1]
```diff
@@ -83,6 +98,32 @@ def __init__(
...
+                logger.warning(
+                    "You have requested that the pipeline use dates "
+                    "provided by the `dates` pipeline, but it was not set. "
+                    "Skipping that step."
+                )
+            elif exclude_birthdate:
+                logger.info(
+                    "You have requested that the pipeline use date "
+                    "and exclude birth dates. "
+                    "To take the most of the feature, "
```
Comment (percevalw) [inline edsnlp/pipelines/qualifiers/history/history.py:115]:
```suggestion
                    "To make the most of this feature, "
```

File: edsnlp/pipelines/qualifiers/history/history.py [hunk 2]
```diff
@@ -83,6 +98,32 @@ def __init__(
...
+                    "make sur you provide the `birth_datetime` "
+                    "context and `note_datetime` context. "
+                )
+            else:
+                logger.info(
+                    "You have requested that the pipeline use date "
+                    "To take the most of the feature, "
```
Comment (percevalw) [inline edsnlp/pipelines/qualifiers/history/history.py:122]:
```suggestion
                    "To make the most of this feature, "
```

File: edsnlp/pipelines/qualifiers/history/history.py [hunk 3]
```diff
@@ -83,6 +98,32 @@ def __init__(
...
+                    "make sur you provide the `note_datetime` "
```
Comment (percevalw) [inline edsnlp/pipelines/qualifiers/history/history.py:123]:
```suggestion
                    "make sure you provide the `note_datetime` "
```

File: edsnlp/pipelines/qualifiers/history/history.py [hunk 4]
```diff
@@ -155,6 +196,9 @@ def set_extensions(cls) -> None:
         if not Span.has_extension("history_cues"):
             Span.set_extension("history_cues", default=[])
 
+        if not Span.has_extension("recent_cues"):
+            Span.set_extension("recent_cues", default=[])
+
```
Comment (percevalw) [inline edsnlp/pipelines/qualifiers/history/history.py:201]: I'm not sure about the `recent_cues` extension, couldn't we just ignore a date completely if it doesn't pass the timedelta requirement? Or if we need the date to explain why a atcd marker was not enough to qualify an entity as history, could we instead group all dates under a single `date_cues` ? What do you think ?

**Review by percevalw — COMMENTED** [review]
Great PR, thanks ! I've left a few comments

File: edsnlp/pipelines/qualifiers/history/history.py (hunk 4 above)
Comment (Aremaki) [reply edsnlp/pipelines/qualifiers/history/history.py:202]: Just added an explanatory comment:

```python
# Store history mentions responsible for the history entity's character
if not Span.has_extension("history_cues"):
    Span.set_extension("history_cues", default=[])

# Store recent mentions responsible for the non-antecedent entity's character
if not Span.has_extension("recent_cues"):
    Span.set_extension("recent_cues", default=[])
```

**Review by percevalw — APPROVED** [review]
Great !

**Pull request merged**